src/kOS/Module/kOSVesselModule.cs

```diff
@@ -223,16 +226,55 @@ private void HookEvents()
         /// </summary>
         private void UnHookEvents()
         {
-            if (RemoteTechHook.IsAvailable(parentVessel.id))
+            if (hasRemoteTech)
             {
-                RemoteTechHook.Instance.RemoveSanctionedPilot(parentVessel.id, UpdateAutopilot);
+                UnHookRemoteTechPilot();
             }
             else
             {
-                parentVessel.OnPreAutopilotUpdate -= UpdateAutopilot;
+                UnHookStockPilot();
             }
         }
 
+        private void CheckSwapEvents()
```

Comment:
I think it's a good idea to mention the reason we're doing this in a comment for this method.  (i.e. comments that are explaining what you're doing might be just a way to excuse badly written code, but commends that are explaning WHY you're doing it, other other hand, are good).

(i.e. mention the race condition here.  Point out that we have no guarantee whether RemoteTech wakes up before or after we do, and therefore `RemoteTechIs.Aavailable` may be false the first time we check this, but then become true a few moments later.)